src/linker/Linker/DirectoryAssemblyResolver.cs

```diff
@@ -50,7 +50,7 @@ public virtual AssemblyDefinition Resolve (AssemblyNameReference name, ReaderPar
 			if (name == null)
```

Comment:
Aren't the ArgumentNullException check now redundant?

Reply:
They're redundant in the sense that the base class performs the same check, but also necessary to prevent passing null to SearchAssemblyPaths (renamed to ResolveFromReferences).
You're right that we already deref name in Resolve,  so maybe that's not strictly necessary, but I still feel that it's safer to keep both checks (moved to the beginning of Resolve).